content/doc/book/managing/system-properties.adoc

```diff
@@ -1319,6 +1319,27 @@ properties:
     Development-mode only: Set to `true` to not skip showing the setup wizard during Jenkins development.
     This property is only effective the first time you run Jenkins in given `JENKINS_HOME`.
 
+- name: jenkins.install.SetupWizard.adminInitialApiToken
+  tags:
+  - security
+  - packaging
+  def: undefined
+  since: 2.260
+  description: |
+    This property determines the behavior during the SetupWizard install phase concerning the API Token creation for the initial admin account.
+    The behavior depends on the provided value:
+
+    `true`:: A token is generated using random value at startup and the information is put in the file `$JENKINS_HOME/secrets/initialAdminApiToken`. This approach is the more interesting (+ secure) when you have read access to the Jenkins home directory.
+    "token" in plain text:: A fixed API Token will be created for the user with provided value as the token.
+    "@[file-location]" which contains plain text value of the token:: A fixed API Token will be created for the user with the value read from the file. The application will not delete the file after read, so the script is responsible to clean up the stuff
+
+    Token format is `[2-char hash version][32-hex-char of secret]`, where the hash version is currently only 11. E.g. 110123456789abcdef0123456789abcdef.
+
+    When the API Token is generated using this system property, it's strongly recommended that you are revoking it during your installation script using the other ways at your disposal so that you have a fresh (random) token with less traces for your script. See https://javadoc.jenkins.io/jenkins/security/ApiTokenProperty.html#generateNewToken-java.lang.String-[ApiTokenProperty#generateNewToken(String)] and https://javadoc.jenkins.io/jenkins/security/ApiTokenProperty.html#revokeAllTokensExceptOne-java.lang.String-[ApiTokenProperty#revokeAllTokensExceptOne(String)] for scripting methods or using the web API calls:
```

Comment:
I'm not understanding the recommendation to revoke the API token during installation.  Is it recommended that a single API token is all that is needed or is it recommended that most of the time there is no API token defined?
```suggestion
    When the API Token is generated using this system property, it should be revoked during the installation script using the other ways at your disposal so that you have a fresh (random) token with less traces for your script. See https://javadoc.jenkins.io/jenkins/security/ApiTokenProperty.html#generateNewToken-java.lang.String-[ApiTokenProperty#generateNewToken(String)] and https://javadoc.jenkins.io/jenkins/security/ApiTokenProperty.html#revokeAllTokensExceptOne-java.lang.String-[ApiTokenProperty#revokeAllTokensExceptOne(String)] for scripting methods or using the web API calls:
```